thunder/dynamo/report.py

```diff
@@ -166,3 +198,235 @@ def print_sorted_memory_info(compute_t: str):
+        # way to set that with print_readable.
+        readable = _readable(self.graph, "DynamoModule", print_output=False)
+
+        COMMAND_LINE_ARGS = """
+import argparse
+
+parser = argparse.ArgumentParser(description="Script for executing an FX graph with specified configurations.")
+
+parser.add_argument(
+    "--check_consistency",
+    type=bool,
+    default=False,
+    help="Whether to check consistency (default: False)"
+)
+parser.add_argument(
+    "--compute_type",
+    type=str,
+    choices=["forward", "forward+backward"],
+    default="forward",
+    help="Type of computation to perform (forward, forward+backward)"
+)
+
+args = parser.parse_args()
+compute_type = args.compute_type
+check_acc = args.check_consistency
+"""
+
+        comment_str = f'''"""
+Environment information get from `torch.utils.collect_env.get_pretty_env_info()`:
+{torch_env}
+
+Versions of Thunder related libraries:
+{thunder_pkgs}"""
+'''
+        del torch_env, thunder_pkgs
+
+        code_str = ""
+        # The packages that are likely to be used by the code generated from the Torch GraphModule
+        code_str += "\n".join([v.import_str for v in torch.fx.graph._custom_builtins.values()]) + "\n"
+        code_str += "\n".join(custom_import_str) + "\n"
+        code_str += f"\n{COMMAND_LINE_ARGS}"
+        func_str = f"def test_{self.graph_name}():\n{readable}\n"
+
+        if any(arg is None for arg in inputs):
+            func_str += f"# Warning: The inputs that cannot be inferred are set to None, requiring the user to manually give inputs according to the code\n"
+        if save_input_tensor:
+            example_inputs = eval(f"[\n{chr(10).join(arg_like(a) for a in inputs)}]")
+            input_file_name = folder / f"{self.graph_name}_inputs.pt"
+            torch.save(example_inputs, input_file_name)
+            func_str += f"inputs = torch.load('{input_file_name}')\n"
+        else:
+            input_str = f"""inputs = [\n{chr(10).join(arg_like(a) for a in inputs)}\n"""
+            func_str += f"{_addindent(input_str, 4)}\n]\n"
+            del input_str
+
+        func_str += f"""
+model = DynamoModule()
+from thunder.dynamo.report import run_repro1
+
+{custom_executor_str}
+result = run_repro1(compiled_model, compute_type, *inputs)
+if check_acc:
+    eager_result = run_repro1(model, compute_type, *inputs)
+    for (compute_t, eager_v), (_, cur_v) in zip(eager_result.items(), result.items()):
+        torch.testing.assert_close(eager_v, cur_v, msg=lambda e : f'{{compute_t}}: {{e}}')
+"""
+
+        with open(folder / f"{self.graph_name}_{custom_executor_name}.py", "w") as f:
+            print(comment_str, file=f)
+            print(code_str, file=f)
+            print(_addindent(func_str, 4), file=f)
+            print(f"\ntest_{self.graph_name}()", file=f)
+
+
+class FXReport:
+    def __init__(self, graph_report_cls: type[FXGraphReportBase], graphs: list[torch.fx.GraphModule]):
```

Comment:
Comment string for this class

Reply:
Following the above thinking about merging the `FXGraphReport` classes, I think this can just accept a list of fx graphs, and not have to take a class argument